### PR TITLE
Literally just adds felinid bans

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

image

## Why It's Good For The Game

image

### CODE REVIEW

**Comment by SomethingFish** [comment]
speedmerge, this will fix the whole game (real)

**Review by SgtHunk — CHANGES_REQUESTED** [review]
The felinid overwrite needs to be on its own file rather than on the kiss overwrite file

**Comment by Greniza** [comment]
You should put this upstream

**Comment by JohnFulpWillard** [comment]
> The felinid overwrite needs to be on its own file rather than on the kiss overwrite file

I put it there because I haven’t tested this yet and every time I open tgs I get ‘reloading watchdog’ or some shit and it doesn’t let me fucking connect

**Comment by Nobelium-XVIII** [comment]
+1

File: fulp_modules/Z_edits/race_bans/races.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/datum/species/human/felinid/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	. = ..()
+	if(is_banned_from(C.ckey, RACE_FELINID))
+		if(prob(50))
+			C.set_species(/datum/species/beefman)
+		else
+			C.set_species(/datum/species/human)
+
+/datum/species/ethereal/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
```
Comment (SpaceDragon00) [inline fulp_modules/Z_edits/race_bans/races.dm:9]: Please atomize your PRs, thank you. This is about felinids, not ethereals

File: fulp_modules/Z_edits/race_bans/races.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/datum/species/human/felinid/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	. = ..()
+	if(is_banned_from(C.ckey, RACE_FELINID))
+		if(prob(50))
```
Comment (SpaceDragon00) [inline fulp_modules/Z_edits/race_bans/races.dm:4]: Code duplication make it a define

File: fulp_modules/Z_edits/race_bans/races.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply fulp_modules/Z_edits/race_bans/races.dm:9]: dont care

File: fulp_modules/Z_edits/race_bans/races.dm [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+			C.set_species(/datum/species/beefman)
+		else
+			C.set_species(/datum/species/human)
+
+/datum/species/ethereal/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	. = ..()
+	if(is_banned_from(C.ckey, RACE_ETHEREAL))
+		if(prob(50))
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply fulp_modules/Z_edits/race_bans/races.dm:12]: dont care

File: fulp_modules/Z_edits/race_bans/races.dm (hunk 2 above)
Comment (JohnFulpWillard) [reply fulp_modules/Z_edits/race_bans/races.dm:4]: dont care

**Comment by ATH1909** [comment]
> "Literally just adds felinid bans"

> Also adds ethereal bans

https://www.youtube.com/watch?v=dTRKCXC0JFg

**Comment by JohnFulpWillard** [comment]
> 
> 
> > "Literally just adds felinid bans"
> 
> > Also adds ethereal bans
> 
> https://www.youtube.com/watch?v=dTRKCXC0JFg

refer to my replies to SpaceDragon00's reviews

**Pull request merged**